src/datatrove/pipeline/tokens/tokenizer.py

```diff
@@ -197,14 +204,35 @@ def save_final_metadata(self, tokenizer_name: str | None = None, token_count: in
 
 
 class DocumentTokenizer(PipelineStep):
+    """ Tokenize the documents in the pipeline using the HuggingFace fast tokenizers library.
+        This pipeline step saves the tokenized documents locally/remotely in a set of files and optionally shuffles documents inside each file.
+
+        You can use a DocumentTokenizerMerger to merge the tokenized files into a single file while also shuffling the file order.
+    
+    Args:
+        output_folder (DataFolderLike): the output folder where to save the tokenized documents
+        local_working_dir (str | None): a local working directory to use for temporary files (before internal shuffling)
+            if None we shuffle in output_folder (can be very slow if it's a remote location) 
+        save_filename (str): the filename to use for the final tokenized files (default: None – use the default filename)
+        tokenizer_name (str): the name of the tokenizer to use, from the HuggingFace tokenizers library (default: "gpt2")
+        eos_token (str): whether to add the EOS token after each document (default: "<|endoftext|>")
+        save_loss_metadata (bool): save the loss information (default: False)
+        shuffle (bool): whether to shuffle documents in the dataset (default: True)
+        batch_size (int): batch size for tokenization (default: 1000)
+        seed (int): the seed to use for shuffling
+        save_final_metadata (bool): whether to save the final metadata (default: True)
+        upload_block_size (int | None): the fsspec size of the upload block for remote filesystems (S3)
+            You can set this if your s3 uploads are failing because of "Part number must be an integer between 1 and 10000, inclusive".
+            Example: 20 * 2**20 (20MB)
+    """
     name = "✍️ Writer"
     type = "🔢 - TOKENIZER"
     _requires_dependencies = ["tokenizers"]
 
     def __init__(
         self,
         output_folder: DataFolderLike,
-        local_working_dir: str | None = None,
+        local_working_dir: Optional[Tuple[DataFolderLike, str]],
```

Comment:
Why `Optional`? `|` is from Python 3.10 which is our minimum version. Also, it should be a `DataFolderLike` and not a tuple